mvp_cd/deploy_app/views.py

```diff
@@ -1,118 +1,159 @@
+    build_info = BuildInfo.objects.all().order_by('-id')[:10]
     return render_to_response(
-        'build_info.html', {'build_info_item': build_info[0]})
-
-
-def deploy_celery_staging(request, ip, pem_path, deployment_path):
-    process_status = subprocess.call(
-        ['bash', 'deploy_staging_celery.sh', pem_path, ip, deployment_path])
-    process_status_staging_celery(request, process_status)
-
-
-def deploy_celery_production(request, ip, pem_path, deployment_path):
-    process_status = subprocess.call(
-        ['bash', 'deploy_prod_celery.sh', pem_path, ip, deployment_path])
-    process_status_production_celery(request, process_status)
-
-
-def deploy_app_staging(request, ip, pem_path, deployment_path):
-    process_status = subprocess.call(
-        ['bash', 'deploy_staging.sh', pem_path, ip, deployment_path])
-    process_status_staging(request, process_status)
-
-
-def deploy_app_production(request, ip, pem_path, deployment_path):
-    process_status = subprocess.call(
-        ['bash', 'deploy_prod.sh', pem_path, ip, deployment_path])
-    process_status_production(request, process_status)
-
-
-def process_status_staging(request, process_status):
-    if process_status == 0:
-        build_info_item = StagingBuildInfo(
-            curr_build_info=json.dumps(request.data['payload']),
-            is_success='True')
-        build_info_item.save()
-    else:
-        build_info_item = StagingBuildInfo(
-            curr_build_info=json.dumps(request.data['payload']),
-            is_success='False')
-        build_info_item.save()
-
-
-def process_status_production(request, process_status):
-    if process_status == 0:
-        build_info_item = ProductionBuildInfo(
-            curr_build_info=json.dumps(request.data['payload']),
-            is_success='True')
-        build_info_item.save()
-    else:
-        build_info_item = ProductionBuildInfo(
-            curr_build_info=json.dumps(request.data['payload']),
-            is_success='False')
-        build_info_item.save()
+        'build_list.html', {'build_list': build_info})
 
 
-def process_status_production_celery(request, process_status):
-    if process_status == 0:
-        build_info_item = ProductionWorkerBuildInfo(
-            curr_build_info=json.dumps(request.data['payload']),
-            is_success='True')
-        build_info_item.save()
-    else:
-        build_info_item = ProductionWorkerBuildInfo(
-            curr_build_info=json.dumps(request.data['payload']),
-            is_success='False')
-        build_info_item.save()
-
-
-def process_status_staging_celery(request, process_status):
+def build_detail_view(request, id):
+    build_info = get_object_or_404(BuildInfo, id=id)
+    serverbuildinfo_list = build_info.serverbuildinfo_set.all().order_by('-id')
+    return render_to_response(
+        'build_info.html',
+        {'build_info_item': build_info,
+            'serverbuildinfo_list': serverbuildinfo_list})
+
+
+def deploy_celery_staging(
```

Comment:
`deploy_app_production, deploy_app_staging, deploy_celery_production` can be written as single functions `deploy`. Variations can be passed as parameters.